src/crepe/manager/LoopTimerManager.h

```diff
@@ -0,0 +1,173 @@
+	 */
+	void set_time_scale(double time_scale);
+
+	/**
+	 * \brief Get the fixed delta time in seconds without scaling by the time scale.
+	 *
+	 * This value is used in the LoopManager to determine how many times 
+	 * the fixed_update should be called within a given interval.
+	 * This value is also the timing value which is used in the fixed_loop to convert pixels to time.
+	 *
+	 * \return The unscaled fixed delta time in seconds.
+	 */
+	double get_fixed_delta_time() const;
+
+	/**
+	 * \brief Set the fixed_delta_time in seconds.
+	 * 
+	 * \param seconds fixed_delta_time in seconds.
+	 * 
+	 * The fixed_delta_time value is used to determine how many times per second the fixed_update and process_input functions are called.
+	 * This value is also the timing value which is used in the fixed_loop to convert pixels to time.
+	 */
+	void set_fixed_delta_time(double seconds);
+
+	/**
+	 * \brief Retrieves the scaled fixed delta time in seconds.
+	 *
+	 * The scaled fixed delta time is the timing value used within the `fixed_update` function. 
+	 * It is adjusted by the time_scale to account for any changes in the simulation's 
+	 * speed.
+	 *
+	 * \return The fixed delta time, scaled by the current time scale, in seconds.
+	 */
+	double get_scaled_fixed_delta_time() const;
+
+private:
+	//! Friend relation to use start,enforce_frame_rate,get_lag,update,advance_fixed_update.
+	friend class LoopManager;
+
+	/**
+	 * \brief Start the loop timer.
+	 *
+	 * Initializes the timer to begin tracking frame times.
+	 */
+	void start();
+	/**
+	 * \brief Enforce the frame rate limit.
+	 *
+	 * Ensures that the game loop does not exceed the target FPS by delaying frame updates as
+	 * necessary.
+	 */
+	void enforce_frame_rate();
+	/**
+	 * \brief Get the accumulated lag in the game loop.
+	 *
+	 * Lag represents the difference between the target frame time and the actual frame time,
+	 * useful for managing fixed update intervals.
+	 *
+	 * \return Accumulated lag in seconds.
+	 */
+	double get_lag() const;
+
+	/**
+	 * \brief Update the timer to the current frame.
+	 *
+	 * Calculates and updates the delta time for the current frame and adds it to the cumulative
+	 * game time.
+	 */
+	void update();
+
+	/**
+	 * \brief Progress the elapsed fixed time by the fixed delta time interval.
+	 *
+	 * This method advances the game's fixed update loop by adding the fixed_delta_time 
+	 * to elapsed_fixed_time, ensuring the fixed update catches up with the elapsed time.
+	 */
+	void advance_fixed_elapsed_time();
+
+private:
+	//! Target frames per second
+	int target_fps = 60;
+	//! Actual frames per second
+	int actual_fps = 0;
+	//! Time scale for speeding up or slowing down the game (0 = pause, < 1 = slow down, 1 = normal speed, > 1 = speed up)
+	double time_scale = 1;
+	//! Maximum delta time in seconds to avoid large jumps
+	std::chrono::duration<double> maximum_delta_time{0.25};
```

Comment:
Ik vindt een floating-point `std::chrono::duration` prima voor tijdsverschillen, omdat die altijd redelijk klein blijven, maar ik zag dat je de totale verlopen tijd returnet als floating-point type, terwijl ik daar juist een `std::chrono::duration<unsigned long long>` oid zou verwachten. Als je deze types vaker nodig hebt kun je ze beter in een `typedef` zetten voor consistentie.

Reply:
Ik dacht dat we hadden besproken dat alles in seconden wordt gereturned in kan erg kleine values niet in seconden returnen. als ik voor de totale verlopen tijd een unsigned long long ga gebruiken wordt het opgeslagen als microseconden.